app/main.py

```diff
@@ -54,24 +62,35 @@ def action_choose_pdf_file_handler(self) -> None:
     def action_choose_public_key_handler(self) -> None:
         self.add_log(self.choose_file("Public key files(*.pem)"))
 
-    def find_private_key_path(self) -> None:
+    def check_pendrive(self) -> None:
         pen_drives = self.detector.find_all_pen_drives()
 
         if len(pen_drives) != 0:
-            self.add_log("Pendrive has been found")
-            for pen_drive in pen_drives:
-                self.add_log(pen_drive)
-
-            pen_drive = self.detector.find_pen_drive_with_private_key(pen_drives)
-
-            if pen_drive is None:
-                self.add_log("Private key has not been found")
-
-            if pen_drive is not None:
-                self.add_log("Private key has been found")
-                print(self.detector.get_private_key_path(pen_drive))
-
-
+            if self.private_key.get_private_key() is None:
+                self.add_log("Pendrive has been detected")
+
+                pen_drive = self.detector.find_pen_drive_with_private_key(pen_drives)
+
+                if pen_drive is None:
+                    self.add_log("Private key has not been found")
+                    self.private_key.reset_private_key()
+
+                if pen_drive is not None:
+                    if self.private_key.get_private_key() is None:
+                        self.add_log("Private key has been found")
+
+                        dlg = PasswordDialog()
+                        if dlg.exec() == QDialog.DialogCode.Accepted:
+                            password = dlg.get_password()
+                            print(password)
+                            self.private_key.load_private_key(
+                                self.detector.get_private_key_path(pen_drive), password
+                            )
+
+                        print(self.private_key.get_private_key())
```

Comment:
this print also doesn't give  any meaningful information
![image](https://github.com/user-attachments/assets/a3b04182-634b-43e1-85f6-4d2c7d28e3e3)